src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -3182,6 +3182,172 @@ def multiply_by_conjugate(self, J):
         R = self.quaternion_algebra()
         return R.ideal(basis, check=False)
 
+    def pushforward(self, J, side=None):
+        """
+        Compute the ideal which is the pushforward of ``self`` through an ideal ``J``.
+        Uses Lemma 2.1.7 of [Ler2022]_. Only works for integral ideals.
```

Comment:
```suggestion
    def pushforward(self, J, side=None):
        r"""
        Compute the ideal which is the pushforward of ``self`` through an ideal ``J``.

        Uses Lemma 2.1.7 of [Ler2022]_. Only works for integral ideals.
```
This method has no examples, only tests. There should be at least one example, if not more.